### PR TITLE
Update igenomes config

### LINKED ISSUE #68: Not a valid path value: 'null'
### Description of the bug

Unable to complete any workflow except the "test" one. 

### Command used and terminal output

```console
nextflow run nf-core/circrna -profile test_full,docker -r dev

Error executing process > 'NFCORE_CIRCRNA:CIRCRNA:CIRCRNA_DISCOVERY:CIRCEXPLORER2_REFERENCE (null)'

Caused by:  Not a valid path value: 'null'
```

### Relevant files

[nextflow.log](https://github.com/nf-core/circrna/files/10314782/nextflow.log)

### System information

- N E X T F L O W  ~  version 22.10.4                  
- Desktop
- local     
- docker
- Ubuntu 22 LTS
- nf-core/circrna v1.0.0-g51abcba

### PR DESCRIPTION
**Pull request by nictru**
I updated the igenomes config because there were some issues concerning `null` values when using igenomes, probably because some of the attributes were missing in the old version

There are some other things fixed:
- Version dumping failed because of unexpected indentation
- Pipeline parameters were overwritten by the default values when set via igenomes
- Linting errors

### CODE REVIEW

**Comment by nictru** [comment]
I figured out that removing the gtf default value was not the smartest idea, since the nf-core linting is failing now. I'll have a second look at this.

File: nextflow.config [hunk 1]
```diff
@@ -34,7 +34,6 @@ params {
     hisat2_build_memory        = '200.GB'
     segemehl                   = null
     star                       = null
-    gtf                        = null
```
Comment (FriederikeHanssen) [inline nextflow.config:37]: you probably have something in the schema json that is not in the nextflow.config

Comment (nictru) [reply nextflow.config:37]: Thanks for the hint - will have a look at it :)

**Comment by maxulysse** [comment]
I feel that you have a template update at the same time.
Can you have a look at https://github.com/nf-core/circrna/pull/78 before?

**Comment by nictru** [comment]
Thanks for the hint @maxulysse. I rebased #78 so that we can merge that one first.

**Comment by maxulysse** [comment]
> Thanks for the hint @maxulysse. I rebased #78 so that we can merge that one first.

sorry, read the other PR before...
Figured you wanted to merge #78 and then use #77 to fix the tests, not the other way around..

:facepalm:

File: modules/local/targetscan/database/main.nf [hunk 2]
```diff
@@ -5,7 +5,7 @@ process TARGETSCAN_DATABASE {
     conda "conda-forge::sed=4.7"
     container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
         'https://depot.galaxyproject.org/singularity/ubuntu:20.04' :
-        'ubuntu:20.04' }"
+        'registry.hub.docker.com/library/ubuntu:20.04' }"
```
Comment (maxulysse) [inline modules/local/targetscan/database/main.nf]:
```suggestion
        'nf-core/ubuntu:20.04' }"
```

Comment (maxulysse) [reply modules/local/targetscan/database/main.nf]: we mirrored it on quay.io/nf-core

Comment (nictru) [reply modules/local/targetscan/database/main.nf]: Ah great thanks! Was just going through all the occurrences of the same problem

Comment (maxulysse) [reply modules/local/targetscan/database/main.nf]: Figured as much

**Comment by nictru** [comment]
Finally everything should be ready now.

**Review by maxulysse — APPROVED** [review]

File: conf/test_cache.config [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    fasta = "${params.test_data_base}/reference/chrI.fa"
+    gtf = "${params.test_data_base}/reference/chrI.gtf"
+    mature = "${params.test_data_base}/reference/mature.fa"
+    tool = "circexplorer2"
+    phenotype = "${params.test_data_base}/phenotype.csv"
+    skip_trimming = false
+    module = "circrna_discovery,mirna_prediction,differential_expression"
+    outdir = "results/"
+    bsj_reads = 2
+    species = "cel"
```
Comment (FriederikeHanssen) [inline conf/test_cache.config]: can you align this by `=` for readability sake?

File: main.nf [hunk 4]
```diff
@@ -17,10 +17,14 @@ nextflow.enable.dsl = 2
...
-//   This is an example of how to use getGenomeAttribute() to fetch parameters
-//   from igenomes.config using `--genome`
-params.fasta = WorkflowMain.getGenomeAttribute(params, 'fasta')
+params.fasta   = WorkflowMain.getGenomeAttribute(params, 'fasta')
+params.gtf     = WorkflowMain.getGenomeAttribute(params, 'gtf')
+params.bwa     = WorkflowMain.getGenomeAttribute(params, 'bwa')
+params.star    = WorkflowMain.getGenomeAttribute(params, 'star')
+params.bowtie  = WorkflowMain.getGenomeAttribute(params, 'bowtie')
+params.bowtie2 = WorkflowMain.getGenomeAttribute(params, 'bowtie2')
+params.mature  = WorkflowMain.getGenomeAttribute(params, 'mature')
```
Comment (FriederikeHanssen) [inline main.nf:26]: mature was removed from the igenomes file, was that intended?

Comment (nictru) [reply main.nf:26]: Good catch, no this was not intended

Comment (nictru) [reply main.nf:26]: There were even some more fields missing

File: conf/igenomes.config [hunk 5]
```diff
@@ -12,418 +12,582 @@ params {
...
+            fasta_fai   = "${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/WholeGenomeFasta/genome.fa.fai"
+            bwa         = "${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/version0.6.0/"
+            bowtie      = "${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BowtieIndex/"
+            bowtie2     = "${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/"
+            star        = "${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/STARIndex/"
+            bismark     = "${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/"
+            gtf         = "${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Annotation/Genes/genes.gtf"
+            bed12       = "${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Annotation/Genes/genes.bed"
+            mature      = "${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Annotation/SmallRNA/mature.fa"
+            readme      = "${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Annotation/README.txt"
```
Comment (FriederikeHanssen) [inline conf/igenomes.config]: Readme?

**Review by FriederikeHanssen — APPROVED** [review]
Nice work!

**Pull request merged**